src/AppBundle/Controller/ArticleInfoController.php

```diff
@@ -98,6 +99,39 @@ public function indexAction(Request $request)
     }
 
     /**
+     * Generate ArticleInfo gadget script for use on-wiki. This automatically points the
+     * script to this installation's API. Pass ?uglify=1 to uglify the code.
+     *
+     * @Route("/articleinfo-gadget.js", name="ArticleInfoGadget")
+     * @link https://www.mediawiki.org/wiki/XTools#ArticleInfo_gadget
+     *
+     * @param Request $request The HTTP request
+     * @return Response
+     */
+    public function gadgetAction(Request $request)
+    {
+        $rendered = $this->renderView('articleInfo/articleinfo.js.twig');
+
+        // SUPER hacky, but it works and is safe.
+        if ($request->query->get('uglify') != '') {
+            $rendered = str_replace('$', '\$', trim($rendered));
+            $rendered = str_replace('"', '\"', trim($rendered));
+            $script = "echo \"$rendered\" | tee temp.js >/dev/null && ";
+            $script .= $this->get('kernel')->getRootDir() .
+                '/Resources/node_modules/uglify-es/bin/uglifyjs temp.js --mangle';
+            $script .=  '&& rm temp.js >/dev/null';
+            $process = new Process($script);
+            $process->run();
```

Comment:
Should throw an exception if the process failed, based on `$process->getErrorOutput()`